### ISSUE TITLE
My Jetpack: Tablet size CSS tweaks

### ISSUE THREAD

**Issue by ebinnion** [Automattic/jetpack#35801]
### Impacted plugin

Jetpack, Boost

### Quick summary

While demoing My Jetpack earlier, in a WoA site, but I imagine the issue persists elsewhere, I noticed that there were a few places where some tweaks needed to be made for tablet views.

On the My Jetpack page at around 1072px:

<img width="282" alt="Screenshot 2024-02-20 at 11 46 21 AM" src="https://github.com/Automattic/jetpack/assets/1126811/530d1bb7-1682-41db-942f-f5ae06e0405b">

On the Boost upsell page (/wp-admin/admin.php?page=jetpack-boost#/getting-started) at > 960px:

<img width="589" alt="Screenshot 2024-02-20 at 11 57 05 AM" src="https://github.com/Automattic/jetpack/assets/1126811/71529b5e-e8be-4fcb-ae1c-e0ee1bba55f7">

### Steps to reproduce

1. On site with connected Jetpack
2. Resize browser to about 1072px
3. View the pages mentioned above

### Impact

One

### Available workarounds?

There is no user impact

**Comment by robertsreberski** [comment]
We are going to redesign interstitials and cards altogether soon:
- https://github.com/Automattic/jetpack-roadmap/issues/1120
- https://github.com/Automattic/jetpack-roadmap/issues/1683

**Comment by CodeyGuyDylan** [comment]
TL;DR: I will open up a PR as a result of this issue, but it has little to do with the issues brought up here as they don't seem to be relevant anymore. I'll be iceboxing this issue and creating a new one that more accurately describes what the PR I'm opening does.

That section that was broken on the backup card has been deprecated and no longer shows up. I've checked to make sure it doesn't have any other responsiveness issues and I couldn't find anything. However, the code for the backup section still existed so I've taken this opportunity to clean up that code.

Additionally I discovered an issue in certain circumstances where there would be no description on the backup card, I think this cropped up during the development of the error messages on the backup card so I'm also going ahead and fixing that.

As for the interstitial issue, like Robert said we are redesigning them as we speak so I won't touch that one.

It seems like what I'm doing has really nothing to do with the issue here, so I'm gonna go ahead and icebox it and create a new one that is more accurate to the PR I'll be opening 😄

**Comment by CodeyGuyDylan** [comment]
[Here is the issue](https://github.com/Automattic/jetpack-marketing/issues/1131) i'll be handling the changes I mentioned in
